Expose ContentDisposition in actix-multipart to fix broken doc link

zacps opened the pull request:
The method `Field.content_disposition` returns `ContentDisposition` which is private, but publically exported by `actix-web`.

This PR re-exports `ContentDisposition` and its fields so that they are properly documented in `actix-multipart`.

Fixes #1109

zacps committed:
Expose ContentDisposition in actix-multipart to fix broken doc link

zacps commented:
This could also be solved by exporting `ContentDisposition` in `actix-http`.

JohnTitor commented:
The doc for `actix-multipart` points to `actix-http`, so this should be fixed on the `actix-http`side, I think.

zacps committed:
Revert "Expose ContentDisposition in actix-multipart to fix broken doc link"

This reverts commit e90d71d16cb552cd3e1745646fabcc48e0b4e379.

zacps committed:
Unhide actix-http::header::common docs

These types are used in other exported documented interfaces and create
broken links if not documented.
See `actix_multipart::Field.content_disposition`

zacps commented:
Yeah I agree. It turns out they are actually exported in `actix-http` but have the `doc(hidden)` attribute. Removing that fixes the broken links.

There's also `actix-http::header::shared` which has `doc(hidden)`. I left that because it doesn't affect the specific problem I encountered, but it might be worth evaluating if there's any reason to hide these docs.
